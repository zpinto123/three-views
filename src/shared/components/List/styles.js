import { css } from "styled-components";
import { tokens } from "../../design";

const styles = {
  container: css`
    width: 1000px;
    display: flex;
    align-items: center;
    justify-content: center;
    margin: auto;
    text-align: center;
    flex-direction: column;
    background-color: ${tokens.COLORS.WATER};
  `
};

export default styles;
