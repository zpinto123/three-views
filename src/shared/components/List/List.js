import React, { Component } from "react";
import PropTypes from "prop-types";
import { styleUtils } from "../../utils";
import styles from "./styles";

const { style } = styleUtils;
const Container = style("div", styles.container);

class List extends Component {
  render() {
    const { items, renderItems } = this.props;
    if (!items) return null;
    return <Container>{items.map(data => renderItems(data))}</Container>;
  }
}

List.propTypes = {
  items: PropTypes.array.isRequired,
  renderItems: PropTypes.func.isRequired
};

export default List;
