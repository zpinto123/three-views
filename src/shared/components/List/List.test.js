import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import "jest-styled-components";

import List from "./";

describe("<List />", () => {
  it("Show List correctly", () => {
    const mockCallBack = jest.fn();

    const component = renderer.create(
      <List items={[1, 2, 3, 4]} renderItems={mockCallBack} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
