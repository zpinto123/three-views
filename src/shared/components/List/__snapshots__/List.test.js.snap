// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> Show List correctly 1`] = `
.c0 {
  width: 1000px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: auto;
  text-align: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #f7f9fc;
}

<div
  className="c0"
/>
`;
